modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v5_1_0/test/DDMFieldDDMFieldUpgradeProcessTest.java

```diff
@@ -0,0 +1,215 @@
+import com.liferay.portal.kernel.service.ServiceContext;
+import com.liferay.portal.kernel.service.change.tracking.CTService;
+import com.liferay.portal.kernel.test.rule.AggregateTestRule;
+import com.liferay.portal.kernel.test.rule.DeleteAfterTestRun;
+import com.liferay.portal.kernel.test.util.GroupTestUtil;
+import com.liferay.portal.kernel.test.util.RandomTestUtil;
+import com.liferay.portal.kernel.test.util.ServiceContextTestUtil;
+import com.liferay.portal.kernel.test.util.TestPropsValues;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.util.ListUtil;
+import com.liferay.portal.kernel.util.PortalUtil;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.test.rule.PermissionCheckerMethodTestRule;
+import com.liferay.portal.upgrade.registry.UpgradeStepRegistrator;
+import com.liferay.portal.upgrade.test.util.UpgradeTestUtil;
+
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Paulo Albuquerque
+ */
+@RunWith(Arquillian.class)
+public class DDMFieldDDMFieldUpgradeProcessTest
+	extends BaseCTUpgradeProcessTestCase {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@Before
+	public void setUp() throws Exception {
+		_group = GroupTestUtil.addGroup();
+		ServiceContext serviceContext =
+			ServiceContextTestUtil.getServiceContext();
+
+		DDMForm ddmForm = DDMFormTestUtil.createDDMForm(
+			RandomTestUtil.randomString());
+
+		_ddmStructure = DDMStructureLocalServiceUtil.addStructure(
+			TestPropsValues.getUserId(), _group.getGroupId(), 0,
+			PortalUtil.getClassNameId(DDLRecordSet.class.getName()),
+			"CUSTOM-META-TAGS", RandomTestUtil.randomLocaleStringMap(), null,
+			ddmForm, _ddm.getDefaultDDMFormLayout(ddmForm),
+			StorageType.DEFAULT.toString(), DDMStructureConstants.TYPE_DEFAULT,
+			serviceContext);
+
+		DDLRecordSet ddlRecordSet = _ddlRecordSetLocalService.addRecordSet(
+			TestPropsValues.getUserId(), _group.getGroupId(),
+			_ddmStructure.getStructureId(), null,
+			RandomTestUtil.randomLocaleStringMap(), null,
+			DDLRecordSetConstants.MIN_DISPLAY_ROWS_DEFAULT,
+			DDLRecordSetConstants.SCOPE_DYNAMIC_DATA_LISTS,
+			ServiceContextTestUtil.getServiceContext());
+
+		DDMFormValues ddmFormValues = _createDDMFormValues();
+
+		DDLRecord ddlRecord = _ddlRecordLocalService.addRecord(
+			TestPropsValues.getUserId(), _group.getGroupId(),
+			ddlRecordSet.getRecordSetId(),
+			DDLRecordConstants.DISPLAY_INDEX_DEFAULT, ddmFormValues,
+			serviceContext);
+
+		ddmStorageLink = ddmStorageLinkLocalService.getClassStorageLink(
+			ddlRecord.getDDMStorageId());
+
+		_priority = RandomTestUtil.randomInt();
+
+		_addDDMField("property", _priority);
```

Comment:
@p-albuquerque can you explain why you did this?

Reply:
These values for the DDMField columns are required to the upgrade run with success:
one field need to has the fieldName = "property" [here](https://github.com/liferay/liferay-portal-ee/blob/master/modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_0/DDMFieldUpgradeProcess.java#L64)
I need to set a "known" priority, because this field will be used to update the other field [here](https://github.com/liferay/liferay-portal-ee/blob/master/modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_0/DDMFieldUpgradeProcess.java#L57-L60) and [here](https://github.com/liferay/liferay-portal-ee/blob/master/modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_0/DDMFieldUpgradeProcess.java#L74-L75)

Reply:
Thanks!